doc/dev/engsys.md

```diff
@@ -0,0 +1,73 @@
+# Engineering System Checks
```

Comment:
@seankane-msft  a ton of this is documented [here ](https://github.com/Azure/azure-sdk-for-python/blob/main/doc/eng_sys_checks.md)

We probably want to link to this doc from here.

Reply:
Didn't even see this, I'll move this to expand on the existing docs. Thanks for pointing it out scott

Reply:
When I read this I actually notice that a lot of the little "intro" details that are easily available here are not present in the other doc, so its not like this is pointless 👍 .